libraries/RC_Channel/RC_Channel.cpp

```diff
@@ -1269,29 +1269,24 @@ RC_Channel *RC_Channels::find_channel_for_option(const RC_Channel::aux_func_t op
 // duplicate_options_exist - returns true if any options are duplicated
 bool RC_Channels::duplicate_options_exist()
 {
-    uint8_t auxsw_option_counts[256] = {};
+    uint8_t auxsw_option_counts[uint32_t(RC_Channel::AUX_FUNC::AUX_FUNC_END) / 8 + 1] = {};
     for (uint8_t i=0; i<NUM_RC_CHANNELS; i++) {
         const RC_Channel *c = channel(i);
         if (c == nullptr) {
             // odd?
             continue;
         }
         const uint16_t option = c->option.get();
-        if (option >= sizeof(auxsw_option_counts)) {
+        if (option == 0 || option >= uint16_t(RC_Channel::AUX_FUNC::AUX_FUNC_END)) {
             continue;
         }
-        auxsw_option_counts[option]++;
-    }
-
-    for (uint16_t i=0; i<sizeof(auxsw_option_counts); i++) {
-        if (i == 0) { // MAGIC VALUE! This is AUXSW_DO_NOTHING
-            continue;
-        }
-        if (auxsw_option_counts[i] > 1) {
+        if ((auxsw_option_counts[option / 8] & (1 << (option % 8))) == 0) {
```

Comment:
It's *way* too easy to get this wrong.  Not saying you have - but I wouldn't trust myself with a *quick* review of this.

This is why we have the Bitmask class: https://github.com/ardupilot/ardupilot/master/libraries/AP_Common/Bitmask.h#L27

So instead of 
```
    uint8_t auxsw_option_counts[uint32_t((RC_Channel::AUX_FUNC::AUX_FUNC_END) +7 ) / 8] = {};
```
you have
```
    Bitmask<RC_Channel::AUX_FUNC::AUX_FUNC_END> auxsw_option_used;
```
and then:

```suggestion
        if (auxsw_option_used.get(option)) {
```

and then also use `.set(option)` as appropriate.